feat(magento2): Update Magento to 2.4.6-p4 and it's PHP to 8.2

andytson-inviqa opened the pull request:
Since 2.4.6, elasticsearch 8 has been supported as well (needs `magento/module-elasticsearch-8` composer package added), but Magento Commerce on Cloud uses opensearch instead

andytson-inviqa committed:
Update Magento to 2.4.6-p3 and it's PHP to 8.2

andytson-inviqa commented:
spotted some improvements I could make

andytson-inviqa committed:
Use elasticsearch 8 when magento >= 2.4.6

andytson-inviqa commented:
and 2.4.6-p3 > 2.4.6 in version_compare correct

kierenevans commented on src/magento2/harness.yml:
```diff
@@ -74,7 +74,7 @@ attributes:
+    version: 2.4.6-p3
   mysql:
     image: mariadb
     tag: '10.4'
```
Should we upgrade this to 10.6 to match "on-prem" for https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/system-requirements.html ?

andytson-inviqa replied:
and even switch to opensearch too

andytson-inviqa committed:
Pin bex/behat-magento2-extension symfony/dependency-injection to one it works with

andytson-inviqa committed:
change removed DeclareStrictTypes newlines prop to lines count

andytson-inviqa committed:
Add magento,search_engine configuration to support elasticsearch8

and partial support for opensearch

andytson-inviqa committed:
Update mariadb to 10.6 for version 2.4.6+

andytson-inviqa committed:
Update config.php for Magento 2.4.6

andytson-inviqa committed:
Exclude only vendor top-level Test folders from classmap

PHPUnit 10 relies on classmap, and no way to re-incude it in autoload-dev due to not being PSR-0 or PSR-4 compatible

andytson-inviqa committed:
Update magento2 version to 2.4.6-p4

andytson-inviqa committed:
Pin further symfony packages to satisfy bex

We need to consider how we can replace it

andytson-inviqa committed:
Update config.php

andytson-inviqa committed:
Update searchengine to pick opensearch by default and elasticsearch version 8 if changed

andytson-inviqa committed:
Enable opensearch service if searchengine.platform is set to it

andytson-inviqa commented on src/magento2/application/overlay/phpcs.xml:
```diff
@@ -59,7 +59,7 @@
     <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
         <properties>
-            <property name="newlinesCountBetweenOpenTagAndDeclare" value="2"/>
+            <property name="linesCountBeforeDeclare" value="1"/>
```
as this is an overlay, projects would need to upgrade slevomat/coding-standard or revert in an override
